### PR TITLE
Presentation docs selectednodeinstances

### PR DESCRIPTION
**Pull request by karolisab**
Updated documentation for selected node instances specification.

TODO:
- [x] Set target to master branch when the presentation_docs_contentrelatedinstances branch is merged to master
- [x] Rename snags to generic example name (update names from presentation_docs_contentrelatedinstances)

### CODE REVIEW

File: docs/presentation/Content/SelectedNodeInstances.md [hunk 1]
```diff
@@ -1,36 +1,175 @@
...
-  "acceptablePolymorphically": true
-}
+| Name                                                                | Required? | Type                                                                                | Default |
+| ------------------------------------------------------------------- | --------- | ----------------------------------------------------------------------------------- | ------- |
+| *Filtering*                                                         |
+| [`acceptableSchemaName`](#attribute-acceptableschemaname)           | No        | `string`                                                                            | `""`    |
+| [`acceptableClassNames`](#attribute-acceptableclassnames)           | No        | `string[]`                                                                          | `[]`    |
+| [`acceptablePolymorphically`](#attribute-acceptablepolymorphically) | No        | `boolean`                                                                           | `false` |
+| [`onlyIfNotHandled`](#attribute-onlyifnothandled)                   | No        | boolean                                                                             | `false` |
```
Comment (grigasp) [inline docs/presentation/Content/SelectedNodeInstances.md:15]:
```suggestion
| [`onlyIfNotHandled`](#attribute-onlyifnothandled)                   | No        | `boolean`                                                                           | `false` |
```

File: docs/presentation/Content/SelectedNodeInstances.md [hunk 2]
```diff
@@ -1,36 +1,175 @@
...
+| [`calculatedProperties`](#attribute-calculatedproperties)           | No        | `CalculatedPropertiesSpecification[]`                                               | `[]`    |
+| [`propertyCategories`](#attribute-propertycategories)               | No        | `PropertyCategorySpecification[]`                                                   | `[]`    |
+| [`propertyOverrides`](#attribute-propertyoverrides)                 | No        | `PropertySpecification[]`                                                           | `[]`    |
+| [`showImages`](#attribute-showimages)                               | No        | `boolean`                                                                           | `false` |
+| *Misc.*                                                             |
+| [`relatedInstances`](#attribute-relatedinstances)                   | No        | [`RelatedInstanceSpecification[]`](../Common-Rules/RelatedInstanceSpecification.md) | `[]`    |
+
+### Attribute: `acceptableSchemaName`
+
+>> **Default behaviour:** All schema names accepted
```
Comment (grigasp) [inline docs/presentation/Content/SelectedNodeInstances.md:29]:
```suggestion
> **Default behaviour:** All schema names accepted
```

File: docs/presentation/Content/SelectedNodeInstances.md [hunk 3]
```diff
@@ -1,36 +1,175 @@
...
+
+  | Selected input                  | Result                                                                                                          |
+  | ------------------------------- | --------------------------------------------------------------------------------------------------------------- |
+  | `BisCore.SpatialViewDefinition` | ![Example when selecting "SpatialViewDefinition"](./media/selectednodeinstances-with-spatialviewdefinition.png) |
+  | `Generic.GroupModel`            | ![Example when selecting "GroupModel"](./media/content-empty-table.png)                                         |
+
+
+### Attribute: `acceptableClassNames`
+
+>> **Default behaviour:** All class names accepted
```
Comment (grigasp) [inline docs/presentation/Content/SelectedNodeInstances.md:45]:
```suggestion
> **Default behaviour:** All class names accepted
```

File: docs/presentation/Content/SelectedNodeInstances.md [hunk 4]
```diff
@@ -1,36 +1,175 @@
...
+Specifies whether derived classes of `acceptableClassNames` should be included in the content.
+
+```ts
+[[include:SelectedNodeInstances.AcceptablePolymorphically.Ruleset]]
+```
+
+  | Selected input                  | Result                                                                                                        |
+  | ------------------------------- | ------------------------------------------------------------------------------------------------------------- |
+  | `BisCore.DictionaryModel`       | ![Example when selecting "DictionaryModel"](./media/selectednodeinstances-with-acceptablepolymorphically.png) |
+  | `BisCore.SpatialViewDefinition` | ![Example when selecting "SpatialViewDefinition"](./media/content-empty-table.png)                            |
```
Comment (grigasp) [inline docs/presentation/Content/SelectedNodeInstances.md:71]: I think a better example would be to have a ruleset with `acceptableClassNames` pointing to a class that has instances of itself and its subclass (not sure if our test imodels have such a case, though). Then we could pass 2 instance keys for the example - one for the class instance and one for the subclass instance. When using `acceptablePolymorphically: true`, content for both instances would be returned. When using `acceptablePolymorphically: false`, only content for the base class instance would be returned.

File: full-stack-tests/presentation/src/learning-snippets/Content.test.ts [hunk 5]
```diff
@@ -893,6 +893,113 @@ describe("Learning Snippets", () => {
...
+        await initialize();
+        imodel = await SnapshotConnection.openFile("assets/datasets/Properties_60InstancesWithUrl2.ibim");
+      });
+
+      afterEach(async () => {
+        await imodel.close();
+        await terminate();
+      });
+
+      it.only("uses `acceptableSchemaName` attribute", async () => {
```
Comment (grigasp) [inline full-stack-tests/presentation/src/learning-snippets/Content.test.ts:910]: `.only` needs to be removed

**Review by grigasp — CHANGES_REQUESTED** [review]

File: docs/presentation/Content/SelectedNodeInstances.md (hunk 4 above)
Comment (karolisab) [reply docs/presentation/Content/SelectedNodeInstances.md]: I tried searching for such case but I did not seem to find one. I did update the example to somewhat match the other content specifications when handling polymorphism.

File: docs/presentation/Common-Rules/MultiSchemaClassesSpecification.md [hunk 6]
```diff
@@ -2,19 +2,19 @@
...
 ## Attributes
 
-| Name                                          | Required? | Type                                                                  | Default |
-| --------------------------------------------- | --------- | --------------------------------------------------------------------- | ------- |
-| [`schemaName`](#attribute-schemaname)         | Yes       | `string`                                                              |   ""    |
-| [`classNames`](#attribute-classnames)         | Yes       | `string[]`                                                            |  `[]`   |
-| [`arePolymorphic`](#attribute-arepolymorphic) | No        | `boolean`                                                             | `false` |
+| Name                                          | Required? | Type       | Default |
+| --------------------------------------------- | --------- | ---------- | ------- |
+| [`schemaName`](#attribute-schemaname)         | Yes       | `string`   | ""      |
```
Comment (grigasp) [inline docs/presentation/Common-Rules/MultiSchemaClassesSpecification.md:11]:
```suggestion
| [`schemaName`](#attribute-schemaname)         | Yes       | `string`   | `""`    |
```

File: docs/presentation/Content/SelectedNodeInstances.md [hunk 7]
```diff
@@ -1,36 +1,175 @@
...
+Specifies whether derived classes of `acceptableClassNames` should be included in the content.
+
+```ts
+[[include:SelectedNodeInstances.AcceptablePolymorphically.Ruleset]]
+```
+
+  | Selecting `BisCore.ViewDefinition` input with `acceptablePolymorphically` set to | Result                                                                                                                                      |
+  | -------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+  | `true`                                                                           | ![Example when selecting "View Definition" with "acceptable polymorphically"](./media/selectednodeinstances-with-spatialviewdefinition.png) |
+  | `false`                                                                          | ![Example when doing a simple "View Definition" select](./media/content-empty-table.png)                                                    |
```
Comment (grigasp) [inline docs/presentation/Content/SelectedNodeInstances.md:71]:
```suggestion
  | `true`                                                                           | ![Example of "acceptable polymorphically" attribute set to "true"](./media/selectednodeinstances-with-spatialviewdefinition.png) |
  | `false`                                                                          | ![Example of "acceptable polymorphically" attribute set to "false"](./media/content-empty-table.png)                                                    |
```

**Review by grigasp — CHANGES_REQUESTED** [review]

File: docs/presentation/Content/SelectedNodeInstances.md (hunk 4 above)
Comment (grigasp) [reply docs/presentation/Content/SelectedNodeInstances.md]: That'll do, thanks

**Review by grigasp — APPROVED** [review]
LGTM. @roluk please review.

**Review by roluk — APPROVED** [review]
Approving everything except hosting proprietary .snag files in this repository.

**Pull request merged**